kpi/utils/object_permission_helper.py

```diff
@@ -43,11 +43,11 @@ def get_assignments_queryset(cls, affected_object, user):
         # Filtering is done in `get_queryset` instead of FilteredBackend class
         # because it's specific to `ObjectPermission`.
         if not user or user.is_anonymous():
-            queryset = queryset.filter(user_id=affected_object.owner.pk)
+            queryset = queryset.filter(user_id=affected_object.owner_id)
         elif not cls.user_can_share(affected_object, user):
             # Display only users' permissions if they are not allowed to modify
             # others' permissions
             queryset = queryset.filter(user_id__in=[user.pk,
-                                                    affected_object.owner.pk])
+                                                    affected_object.owner_id])
```

Comment:
I'm totally in favor of this but am embarrassed to say that I can't find Django documentation to support my observation that `related_field_id` avoids the extra query required for `related_field.pk`. If you see any, please let me know.

Reply:
Couldn't find any either.

Reply:
🤷